superset/models/core.py

```diff
@@ -691,27 +731,30 @@ def _log_query(sql: str) -> None:
         with self.get_raw_connection(catalog=catalog, schema=schema) as conn:
             cursor = conn.cursor()
             df = None
-            for i, sql_ in enumerate(sqls):
-                sql_ = self.mutate_sql_based_on_config(sql_, is_split=True)
-                _log_query(sql_)
-                with event_logger.log_context(
-                    action="execute_sql",
-                    database=self,
-                    object_ref=__name__,
-                ):
-                    self.db_engine_spec.execute(cursor, sql_, self)
-                    if i < len(sqls) - 1:
-                        # If it's not the last, we don't keep the results
-                        cursor.fetchall()
-                    else:
-                        # Last query, fetch and process the results
-                        data = self.db_engine_spec.fetch_data(cursor)
-                        result_set = SupersetResultSet(
-                            data, cursor.description, self.db_engine_spec
-                        )
-                        df = result_set.to_pandas_df()
-            if mutator:
-                df = mutator(df)
+            with temporarily_disconnect_db():
+                if is_feature_enabled("SIMULATE_SLOW_ANALYTICS_DATABASE"):
+                    time.sleep(30)
```

Comment:
nit: would be nice to have a configuration key for this value also, or just remove this functionality and rely on other types of tests, for example using charts that query pg_sleep on a PG analytics db

Reply:
I do think it's a bit squicky to have a production feature flag which inserts a sleep to set up some test scenario most people won't care about. I get that it makes testing easier but it'd probably be better just as a set of test instructions, especially if you can test it with a deliberately slow query like @dpgaspar  suggested

Reply:
I agree with @giftig - adding this type of testing logic is IMO not a great solution, as it convolutes the codebase and adds maintenance burden that doesn't contribute to the core functionality of the product.

Reply:
Yeah it's funky, removing. I mean to issue this PR as DRAFT, let me switch it. Also having this in multiple places isn't great.

Reply:
Also found `SELECT pg_sleep(), {...}` as a different/better way to test this feature.